[PE-D][Tester B] Having `null` attendance catastrophically crashes the app

Given a starting state of the `addressbook.json` file:
```json
{
  "persons" : [ {
    "name" : "Charlotte Oliveiro",
    "studentId" : "A0123456C",
    "email" : "charlotte@u.nus.edu",
    "phone" : "93210283",
    "teleHandle" : "@charlotteo",
    "tutorialGroup" : "T03",
    "attendance" : [ true, false, false, false, false, false, false, false, false, false, false, false, false ]
  } ]
}
```

Suppose a user wants to nullify a student's attendance by editing the `json` file, and changes one week of the attendance to be `null` instead of `false`, such that the file is as such:
```json
{
  "persons" : [ {
    "name" : "Charlotte Oliveiro",
    "studentId" : "A0123456C",
    "email" : "charlotte@u.nus.edu",
    "phone" : "93210283",
    "teleHandle" : "@charlotteo",
    "tutorialGroup" : "T03",
    "attendance" : [ null, false, false, false, false, false, false, false, false, false, false, false, false ]
  } ]
}
```
Upon attempting to run the app, the console throws the exception below, and the app fails to open:
```
Exception in Application init method
Exception in thread "main" java.lang.RuntimeException: Exception in Application init method
        at com.sun.javafx.application.LauncherImpl.launchApplication1(LauncherImpl.java:896)
        at com.sun.javafx.application.LauncherImpl.lambda$launchApplication$2(LauncherImpl.java:196)
        at java.base/java.lang.Thread.run(Thread.java:842)
Caused by: java.lang.NullPointerException: Cannot invoke "java.lang.Boolean.booleanValue()" because the return value of "java.util.List.get(int)" is null
        at seedu.address.storage.JsonAdaptedPerson.toModelType(JsonAdaptedPerson.java:135)
        at seedu.address.storage.JsonSerializableAddressBook.toModelType(JsonSerializableAddressBook.java:51)
        at seedu.address.storage.JsonAddressBookStorage.readAddressBook(JsonAddressBookStorage.java:55)
        at seedu.address.storage.StorageManager.readAddressBook(StorageManager.java:64)
        at seedu.address.storage.StorageManager.readAddressBook(StorageManager.java:58)
        at seedu.address.MainApp.initModelManager(MainApp.java:87)
        at seedu.address.MainApp.init(MainApp.java:68)
        at com.sun.javafx.application.LauncherImpl.launchApplication1(LauncherImpl.java:825)
        ... 2 more
```
This causes the product's support for editing the data files to be [less than that of AB3](https://nus-cs2103-ay2526-s2.github.io/website/admin/tp-ped.html#:~:text=Handling%20manual%20edits,Editable%2DFile), violating `Constraint-Human-Editable-File`

-------------
Labels: `severity.High` `type.FunctionalityBug`

--- a/src/main/java/seedu/address/storage/JsonAdaptedPerson.java
+++ b/src/main/java/seedu/address/storage/JsonAdaptedPerson.java
@@ -132,7 +132,7 @@ class JsonAdaptedPerson {
         Attendance modelAttendance = new Attendance();
         if (!attendance.isEmpty()) {
             for (int i = 0; i < attendance.size() && i < Attendance.MAX_WEEKS; i++) {
-                if (attendance.get(i)) {
+                if (Boolean.TRUE.equals(attendance.get(i))) {
                     modelAttendance.markWeek(i + 1);
                 }
             }

